keanu-project/src/main/java/io/improbable/keanu/kotlin/BooleanOperators.kt

```diff
@@ -0,0 +1,26 @@
+package io.improbable.keanu.kotlin
+
+/**
+ * Custom operators for booleans.
+ *
+ * Built-in operators (&&, ||) cannot be overridden so infix functions are used instead.
```

Comment:
This is interesting, might be worth including a link in the javadoc: https://discuss.kotlinlang.org/t/overloading-and-and-or-operators/1623/2

> Overloading ‘and’ and ‘or’ is recommended against in most languages because short-circuiting is lost, which changes the usual symantics of the operators.

Reply:
This is a really interesting point about short-circuiting. I'll link to the discussion